forge/auth.js

```diff
@@ -17,61 +17,82 @@ module.exports = fp(async function (app, opts, done) {
     })
 
     async function checkToken (projectId, token) {
-        try {
-            await client.get(projectId, {
-                headers: {
-                    authorization: `Bearer ${token}`
-                }
-            })
-            // console.log('token good')
-            return true
-        } catch (err) {
-            // console.log('token bad')
-            return false
-        }
+        const project = await client.get(projectId, {
+            headers: {
+                authorization: `Bearer ${token}`
+            }
+        })
+        return !!project
     }
 
     async function checkAuth (request, reply) {
-        if (request.headers?.authorization) {
-            const parts = request.headers.authorization.split(' ')
-            if (parts.length === 2) {
-                const scheme = parts[0]
-                const token = parts[1]
-                if (scheme !== 'Bearer') {
-                    reply.code(401).send({ code: 'unauthorized', error: 'unauthorized' })
+        try {
+            const token = getAuthToken(request)
+            const cacheOk = checkCache(token, request.params.projectId)
+            if (!cacheOk) {
```

Comment:
Does this `if` need an `else` to send a 401?

Reply:
No Ben, if successful (i.e. `cacheOk`) it simply passes through the function without replying (carries on with route)
This is covered in tests too & I did a lot of manual testing - pretty sure we capture all scenarios.
NOTE: in case you missed it:  `getAuthToken()` will throw in not ok - that leads to a 401 in the catch block.